dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/RateLimiter.java

```diff
@@ -0,0 +1,224 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Bucket;
+import io.github.bucket4j.Refill;
+import io.github.bucket4j.TimeMeter;
+import java.time.Duration;
+import java.util.UnknownFormatConversionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This class uses <a href="https://en.wikipedia.org/wiki/Token_buckete">Token bucket</a>
+ * algorithm to evenly process the available resources in a given amount of time.
+ * <pre>
+ * With every PR sent by DFIU tool, a token will be consumed. This will ensure that a PR will
+ * be raised only when there is a token left in the bucket. Initially bucket will have a fixed
+ * number of tokens equal to the 'rateLimit' value. The bucket will be refilled with tokens at
+ * the end of every 'rateLimitDuration' with 'rateLimit' tokens ensuring that at any point of
+ * time, the number of tokens in the bucket won't exceed the 'rateLimit', limiting the rate at
+ * which PRs would be raised. To keep the refill rate uniform, one token will be added to the
+ * bucket every 'tokenAddingRate'. If no tokens are left in the bucket, then PR won't be raised
+ * and program will halt until next token is available.
+ * <pre>
+ */
+public class RateLimiter {
+
+    private static final Logger log = LoggerFactory.getLogger(RateLimiter.class);
+    private final long rateLimit;
```

Comment:
rateLimit is duplicate with `rateLimitDuration`....can you make a single 

`Duration rateLimit;`

Reply:
rateLimitDuration is already initialized with the rate

Reply:
specifically here: https://github.com/salesforce/dockerfile-image-update/pull/717/files#diff-b17e0a9f060287f7ce7cd459e67a05449d5980ee6faddf7c31e8ac1c6848e6a0R45